### PR TITLE
feat: added filter logic

### PR DESCRIPTION
**Pull request by SanjayJatti**
Files to be reviewed:
- filters.jsx
- ProductPage.jsx
- ProductCard.jsx
- FilterContext.jsx
- ProductsContext.jsx
- FilterReducer.jsx
- ProductReducer.jsx
- Getdata.js
filter-logic

### CODE REVIEW

**Review by KanekarSnehal — APPROVED** [review]
- code and folders are well structured.
- I liked how the similar code is separated into different files
-  Just a suggestion, you can host all the images on cdn and give the links in src attribute

File: src/Components/ProductsPage/Filter.jsx [hunk 1]
```diff
@@ -1,73 +1,115 @@
...
-        <div className="margin-b-sm">
-          {filters.map((filter, index) => {
-            return (
-              <div className="flex-column gap-xs">
+        <div className="flex-column gap-xs">
+          <div className="input-checkbox">
+            <input
+              type="checkbox"
+              id="FILTER_BY_STOCK"
+              name="FILTER_BY_STOCK"
```
Comment (narharikale) [inline src/Components/ProductsPage/Filter.jsx:35]: can we use constants here, it will easier

File: src/Context/ProductsContext.jsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { createContext, useContext, useReducer, useEffect } from "react";
+import { getProducts } from "../Utils/GetData";
```
Comment (narharikale) [inline src/Context/ProductsContext.jsx:2]: we can use barrel export here

**Review by narharikale — APPROVED** [review]

**Comment by SanjayJatti** [comment]
> * code and folders are well structured.
> * I liked how the similar code is separated into different files
> * Just a suggestion, you can host all the images on cdn and give the links in src attribute

Sure will look into it.

**Pull request merged**